AppCenter/AppCenter/MSAppCenter.m

```diff
@@ -192,24 +196,22 @@ - (instancetype)init {
  * Configuring without an app secret is valid. If that is the case, the app secret will
  * not be set.
  */
-- (BOOL)configureWithAppSecret:(NSString *)appSecret {
+- (BOOL)configureWithSecretString:(NSString *)secretString fromApplication:(BOOL)fromApplication {
   @synchronized(self) {
     BOOL success = false;
-    if (self.sdkConfigured) {
+    if (self.configuredFromApplication && fromApplication) {
       MSLogAssert([MSAppCenter logTag], @"App Center SDK has already been configured.");
-    }
-
-    // Validate and set the app secret, if one is provided.
-    else if (appSecret && [appSecret length] == 0) {
-      MSLogAssert([MSAppCenter logTag], @"AppSecret is invalid.");
     } else {
-      self.appSecret = [MSUtility appSecretFrom:appSecret];
-      self.defaultTransmissionTargetToken = [MSUtility transmissionTargetTokenFrom:appSecret];
+      if (!self.appSecret) {
+        self.appSecret = [MSUtility appSecretFrom:secretString];
+      }
+      self.defaultTransmissionTargetToken = [MSUtility transmissionTargetTokenFrom:secretString];
```

Comment:
Do we need to add the null check on the defaultTransmissionTargetToken too?

Reply:
Oh.. I didn't think that's the case but probably yes.